feat: blogpost for 2.7.x (#1758)
kumahq/kuma-website: app/_posts/2024-04-17-kuma-2-7-0.md

---
title: Kuma 2.7 release with GUI improvements, shadow policies and more...
description: Kuma 2.7 release with GUI improvements, shadow policies and more...

> date: 2024-04-17

Reviewer: Not sure how important/necessary this is but if you need to change the date don't forget to do that!

headerImage: /assets/images/blog/test-header.jpg
tags:
  - Release
---

We’re excited to announce the release of Kuma 2.7, as usual the release is packed with new features and improvements.

We've picked these highlights for you:

- GUI improvements for easier management of your service mesh
- GatewayAPI integration is now Generally Available and uses MeshHTTPRoute
- Shadow policies to help you migrate to the new policies
- Better `MeshMetric` OpenTelemetry support and profiles

You can always check our [release notes](https://github.com/kumahq/kuma/releases/tag/2.7.0) for the full list of changes.

## GUI improvements

> List views now displays names, namespaces, and zones for policies and dataplanes, providing a clear understanding of resource placement in multi-zone environments:

Reviewer: ```suggestion
List views now display names, namespaces, and zones for policies and dataplanes, providing a clear understanding of resource placement in multi-zone environments:
```

![List of kuma dataplanes showing the namespace](/assets/images/blog/kuma_2_7_dpp_list.png)

The GUI now offers a dedicated view of routes managed by your built-in gateway, providing a clear overview of your gateway configuration:

![A kuma gateway showing the routes in use](/assets/images/blog/kuma_2_7_gateway_view.png)

The dataplane view now displays policies applied to inbound and outbound traffic, simplifying proxy behavior comprehension:

![A kuma gateway showing the routes in use](/assets/images/blog/kuma_2_7_dpp_policies.png)

## GatewayAPI integration
Our Gateway API integration now uses [`MeshHTTPRoute`](/docs/2.7.x/policies/meshhttproute), enabling us to retire `MeshGatewayRoute` which will be deprecated and removed in the future.
We've also documented how to [use the Gateway API with multi-zone](/docs/2.7.x/using-mesh/managing-ingress-traffic/gateway-api/#multi-zone-deployments).

Additionally, we're thrilled to announce that our entire [Gateway API](https://gateway-api.sigs.k8s.io/) integration, including GAMMA support, is now Generally Available and no longer considered experimental.

## Shadow policies and migration of policies

> Since Kuma 2.0 we're rolling out targetRef policies. These policies are more powerful and easier to use than the previous version.

Reviewer: ```suggestion
Since Kuma 2.0 we've been rolling out targetRef policies. These policies are more powerful and easier to use than the previous version.
```

Not totally sure this what you mean here put pretty sure this is what it should be?

> In 2.6.0 we made these default for new users and in 2.7.0 we're introducing shadow policies to help users migrate from the old policies to the new ones.

Reviewer: ```suggestion
In 2.6.0 we made these the default for new users and in 2.7.0 we're introducing shadow policies to help users migrate from the old policies to the new ones.
```

another "not too sure about", this one might just be personal ways of saying things

This is a great way to test your new policies before enforcing them and to add new policies without breaking your existing setup.

This makes it easier not only to migrate but also for users to figure what would happen if they applied a new policy. 
We have also written a [guide to migrate to new policies](/docs/2.7.x/guides/migration-to-the-new-policies/) which should greatly simplify migration.

## MeshMetric OpenTelemetry support and profiles

In 2.6.0 we introduced [`MeshMetric`](/docs/2.7.x/policies/meshmetric), a powerful `targetRef` based policy to collect and publish metrics from your services and sidecars.

In this release we've improved it further with:
- [OpenTelemetry](/docs/2.7.x/policies/meshmetric/#opentelemetry) data collection supporting multiple backends, more configuration and better reliability.
- [Profiles](/docs/2.7.x/policies/meshmetric/#profiles) to significantly reduce metric volume, lowering observability costs and making it easier to find what you are looking for.

To complement this, we've added a [guide on metrics](/docs/2.7.x/guides/otel-metrics/) to help you get started with `MeshMetric`.

## Upgrading

> We strongly suggest upgrading to Kuma 2.7.0. Upgrading is easy through `kumactl` or Helm.

Reviewer 2: 2.7.1 maybe?

> We strongly suggest upgrading to Kuma 2.7.0. Upgrading is easy through `kumactl` or Helm.

Author: ```suggestion
We strongly suggest upgrading to Kuma 2.7.1. Upgrading is easy through `kumactl` or Helm.
```
